python-checks/src/main/java/org/sonar/python/checks/HardcodedCredentialsCallCheck.java

```diff
@@ -0,0 +1,181 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.StringLiteral;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.tree.TreeUtils;
+
+@Rule(key = "S6437")
+public class HardcodedCredentialsCallCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Revoke and change this password, as it is compromised.";
+  private final Map<String, CredentialMethod> methods;
+
+  public HardcodedCredentialsCallCheck() {
+    methods = new CredentialMethodsLoader().load();
+  }
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, this::processCallExpression);
+  }
+
+  private void processCallExpression(SubscriptionContext ctx) {
+    Optional.of(ctx.syntaxNode())
+      .filter(CallExpression.class::isInstance)
+      .map(CallExpression.class::cast)
+      .filter(this::callHasToBeChecked)
+      .ifPresent(call -> {
+        var method = getMethod(call);
+        method.indices()
+          .forEach(argumentIndex -> {
+            var argumentName = method.args().get(argumentIndex);
+            var argument = TreeUtils.nthArgumentOrKeyword(argumentIndex, argumentName, call.arguments());
+            if (argument != null) {
+              checkArgument(ctx, argument);
+            }
+          });
+      });
+  }
+
+  private static void checkArgument(SubscriptionContext ctx, RegularArgument argument) {
+    var argExp = argument.expression();
+    if (argExp.is(Tree.Kind.STRING_LITERAL)) {
+      Optional.of(argExp)
+        .filter(StringLiteral.class::isInstance)
+        .map(StringLiteral.class::cast)
+        .map(StringLiteral::trimmedQuotesValue)
+        .filter(Predicate.not(String::isEmpty))
+        .ifPresent(value -> ctx.addIssue(argument, MESSAGE));
+    } else if (argExp.is(Tree.Kind.NAME)) {
+      findAssignment((Name) argExp, 0)
+        .filter(StringLiteral.class::isInstance)
+        .map(StringLiteral.class::cast)
+        .filter(string -> Optional.of(string)
```

Comment:
Don't know exactly how it works in Java here. Could we un-nest the map here by flatMapping on `Optional.of` and `.filter(str -> !StringLiteral(str).isEmpty())`. I think it would be more readable without the nested map.